WeatherForecast/WeatherForecast/DetailDateFormattable.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  DetailDateFormattable.swift
+//  WeatherForecast
+//
+//  Created by Tony Lim on 2/4/24.
+//
+
+import Foundation
+
+protocol DetailDateFormattable: AnyObject {
+    
+}
+
+extension DetailDateFormattable {
+    var detailDateFormatter: DateFormatter {
+        let formatter = DateFormatter()
+        formatter.locale = .init(identifier: "ko_KR")
+        formatter.dateFormat = "yyyy-MM-dd(EEEEE) a HH:mm"
+        return formatter
+    }
+}
```

Comment:
```suggestion
protocol DateFormattable: AnyObject {
    func format(date: Date, with dateFormatter: DateFormatter) -> String
}
```

이런 느낌으로 프로토콜을 선언해보는건 어떨까요?

Reply:
따로 함수를 어떻게 지정해줄지 고민하다가 extension을 사용했었는데 저렇게 하면 상황에 맞게 코드를 구현하기 좋겠네요.